dana-bot/src/www.js

```diff
@@ -381,6 +382,27 @@ app.get('/bot/editRepository',
     });
   });
 
+
+function fetchRepo(req, res, repoName, serviceName) {
+    if (!fs.existsSync(globalWWW.config.globalBot.repoPath + '/' + repoName)) {
+        console.log('/bot/' + serviceName + ' -- Cloning repo ' + repoName + ' in ' + globalWWW.config.globalBot.repoPath);
+        gitForBot.setRepoPath(globalWWW.config.globalBot.repoPath);
+        gitForBot.setRepo(globalWWW.config.globalBot.repositories[repoName]);
+        var cloneOut = gitForBot.clone();
+        if (cloneOut.err) {
+          console.log('/bot/' + serviceName + ' -- error while cloning ' + repoName + ': ' + JSON.stringify(cloneOut, null, 2));
+          appError(req, res, '/bot/' + serviceName + ', repository of ' + repoName + ' could not be cloned');
+          return;
+        }
+        var fetchOut = gitForBot.fetch();
```

Comment:
IMO the fetch part should be in the else clause of the if. I mean you only need to 'git fetch' if the repo already exists (and you want to update remote info), otherwise you'd do the 'git clone' which would give an up-to-date repo.

Reply:
I agree, although the fetch part should take no time if the repo just got cloned, it makes more sense that way. Done in upcoming commit.